spec/lib/oauth/client_authentication/none_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Doorkeeper::OAuth::ClientAuthentication::None do
+  describe 'matches_request?' do
+    it "matches if the request doesn't have authorization or client_secret" do
+      request = mock_request({
+        client_id: '1234'
+      })
+
+      expect(described_class.matches_request?(request)).to be true
+    end
+
+    it "doesn't match if the request has client_secret" do
+      request = mock_request({
+        client_id: '1234',
+        client_secret: "5678"
+      })
+
+      expect(described_class.matches_request?(request)).to_not be true
```

Comment:
`matches_request?` actually returns `true` or `nil`, not true or false, just due to a quirk of boolean logic in ruby.